src/rules/no-cycle.js

```diff
@@ -98,6 +106,16 @@ module.exports = {
         return;  // no-self-import territory
       }
 
+      /* If we're in the same Strongly Connected Component,
+       * Then there exists a path from each node in the SCC to every other node in the SCC,
+       * Then there exists at least one path from them to us and from us to them,
+       * Then we have a cycle between us.
+       */
+      const hasDependencyCycle = scc[myPath] === scc[imported.path];
```

Comment:
when `disableScc` is true, won't this always be `false`, meaning it will always early-return on the next line?

Reply:
When `disableScc` is `true`, `scc` will be `{}`, so `scc[whatever]` will be `undefined`.
`undefined === undefined` is always true.

Reply:
even if the path is `'toString'`?

Reply:
Fine. https://github.com/import-js/eslint-plugin-import/pull/2998/commits/9cf7dbc25dc4d85ad154b3cec0542843134f2102